samples/net/sockets/echo_client/src/tcp.c

```diff
@@ -87,16 +87,16 @@ static int compare_tcp_data(struct data *data, const char *buf, uint32_t receive
 	return 0;
 }
 
-static int start_tcp_proto(struct data *data, struct sockaddr *addr,
-			   socklen_t addrlen)
+static int start_tcp_proto(struct data *data, sa_family_t family,
```

Comment:
I don't quite understand the error with this subscript warning. There is family field in sockaddr struct so why can't we use it directly. Should we investigate this more as it looks like there is an error where we define the addr array or something like that.

Reply:
I agree the error dosen't give much sense, I didn't really see anything wrong with the previous code. And actually, googling the error gives some hits about a bug in GCC. However, leaving it as is would break the CI on this PR.
I'm open for better solutions though (or maybe I really did not see something).

Reply:
Ok, lets go with this solution then.

Reply:
I've encountered similar warnings before with sample code using the `sockaddr_storage` struct, but never directly for accessing `addr->sa_family` like this.

I agree its fine to proceed with your fix for now, but this does feel like a problem that should probably be investigated in detail at some point, until we understand why it is happening